tests/bluetooth/audio/mocks/src/conn.c

```diff
@@ -29,3 +29,12 @@ void bt_conn_unref(struct bt_conn *conn)
 {
 
 }
+
+void mock_bt_conn_disconnected(struct bt_conn *conn)
+{
+	STRUCT_SECTION_FOREACH(bt_conn_cb, cb) {
+		if (cb->disconnected) {
+			cb->disconnected(conn, 0x13);
+		}
+	}
+}
```

Comment:
Consider add the reason as a parameter here; we may want to test with both `BT_HCI_ERR_REMOTE_USER_TERM_CONN` and `BT_HCI_ERR_CONN_TIMEOUT`